extensions/2.0/Khronos/KHR_image_ktx2/README.md

```diff
@@ -0,0 +1,74 @@
+# KHR_image_ktx2
+
+## Contributors
+
+* Norbert Nopper, UX3D
+* Alexey Knyazev
+* Members of 3D Formats Working Group
+
+## Status
+
+Draft
+
+## Dependencies
+
+Written against the glTF 2.0 spec.
+
+## Overview
+
+This extension adds the ability to specify images using the [KTX2 file format](http://github.khronos.org/KTX-Specification/) (KTX2). An implementation of this extension can use the images provided in the KTX2 files as an alternative to the PNG or JPG images available in glTF 2.0. Furthermore, specifying images having advanced features such as mip map levels, cube map faces, or arrays is possible.
+
+When the extension is used, it's allowed to use value `image/ktx2` for the `image.mimeType` property.
+
+Optionally, a JSON form of the KTX2 header may be provided in the extension object, so implementations can check whether the image is compatible with the GPU and allocate memory before fetching the actual image data. Also, this allows immediately requesting needed parts (e.g., mip levels) of the KTX2 file by skipping the binary KTX2 header.
+
+Because of glTF compatibility requirements, the image cannot be used directly from any glTF core object, so additional extensions are required to use KTX2 images with glTF textures. The following example shows a complete glTF 2.0 asset containing only an uncompressed RGBA8 2D image with BGRA swizzling metadata and without precomputed mip levels.
+
+```json
+{
+    "asset": {
+        "version": "2.0"
+    },
+    "extensionsUsed": ["KHR_image_ktx2"],
+    "images": [
+        {
+            "uri": "texture.ktx2",
+            "extensions": {
+                "KHR_image_ktx2" : {
+                    "vkFormat": 37,
+                    "pixelWidth": 512,
+                    "pixelHeight": 512,
+                    "levels": [
+                        {
+                            "byteOffset": 256,
+                            "byteLength": 1048576,
+                        }
+                    ],
+                    "metadata": {
+                        "KTXswizzle": "bgra"
+                    }
+                }
+            }
+        }
+    ]
+}
+```
+
+### Validation and implementation notes
+
+If present, the information in the extension object must exactly match KTX2 header.
+
+JSON type of each metadata entry depends on the type of the original value. The JSON-Schema provides types for metadata keys defined in the KTX2 specification.
+
+This extension does not restrict texel formats or other features of the KTX2 images. Such limitations must be defined by other glTF extensions referring to this extension.
```

Comment:
Is this necessary when the extension is marked as required? (I understand that it's possible to replace references to KTX2 images in "texture" object with references through extensions, but I'm not sure I understand why this is better than simply using an `image/ktx2` MIME type on the image objects that are referenced directly)

Reply:
My understanding is that `KHR_image_ktx2` may be skipped entirely, and that the other extensions in this PR would simply independently allow KTX2 images. @lexaknyazev @bghgary is that a correct understanding of https://github.com/KhronosGroup/glTF/pull/1705#discussion_r345996088? This was confusing me as well.

Reply:
(quick clarification, the comment got misplaced and refers to line 65 - implementation note - of the spec, not line 63)

Reply:
> KHR_image_ktx2 may be dropped entirely

This was my suggestion after attempting to implement it in Babylon. Unless we need some other metadata that goes with the extension, it will likely end up with a no-op implementation.